src/ImageSharp/Memory/ByteMemoryManager{T}.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+using System;
+using System.Buffers;
+using System.Runtime.InteropServices;
+
+namespace SixLabors.ImageSharp.Memory
+{
+    /// <summary>
+    /// A custom <see cref="MemoryManager{T}"/> that can wrap <see cref="Memory{T}"/> of <see cref="byte"/> instances
+    /// and cast them to be <see cref="Memory{T}"/> for any arbitrary unmanaged <typeparamref name="T"/> value type.
+    /// </summary>
+    /// <typeparam name="T">The value type to use when casting the wrapped <see cref="Memory{T}"/> instance.</typeparam>
+    internal sealed class ByteMemoryManager<T> : MemoryManager<T>
+        where T : unmanaged
+    {
+        /// <summary>
+        /// The wrapped <see cref="Memory{T}"/> of <see cref="byte"/> instance.
+        /// </summary>
+        private readonly Memory<byte> memory;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ByteMemoryManager{T}"/> class.
+        /// </summary>
+        /// <param name="memory">The <see cref="Memory{T}"/> of <see cref="byte"/> instance to wrap.</param>
+        public ByteMemoryManager(Memory<byte> memory)
+        {
+            this.memory = memory;
+        }
+
+        /// <inheritdoc/>
+        protected override void Dispose(bool disposing)
+        {
+        }
+
+        /// <inheritdoc/>
+        public override Span<T> GetSpan()
+        {
```

Comment:
Just do 
```cs
MemoryMarshal.Cast<byte, T>(memory.Span))
```

(The current logic doesn't work properly for arrays, it needs to return the span of the whole array not the segment iirc)

Reply:
Done in https://github.com/SixLabors/ImageSharp/pull/1314/commits/32380d2bb947737c319ed381d579f34855c0af18, thanks! 👍